pallets/parachain-staking/src/lib.rs

```diff
@@ -1375,6 +1390,86 @@ pub mod pallet {
 
 			Ok(().into())
 		}
+
+		/// Notify a collator is inactive during MaxOfflineRounds
+		#[pallet::call_index(29)]
+		#[pallet::weight(<T as Config>::WeightInfo::notify_inactive_collator())]
+		pub fn notify_inactive_collator(
+			origin: OriginFor<T>,
+			collator: T::AccountId,
+		) -> DispatchResult {
+			ensure!(
+				<EnableMarkingOffline<T>>::get(),
+				<Error<T>>::MarkingOfflineNotEnabled
+			);
+			ensure_signed(origin)?;
+
+			let mut collators_len = 0usize;
+			let max_collators = <TotalSelected<T>>::get();
+
+			if let Some(len) = <SelectedCandidates<T>>::decode_len() {
+				collators_len = len;
+			};
+
+			// Check collators length is not below or eq to 66% of max_collators.
+			// We use saturating logic here with (2/3)
+			// as it is dangerous to use floating point numbers directly.
+			ensure!(
+				collators_len * 3 > (max_collators * 2) as usize,
+				<Error<T>>::TooLowCollatorCountToNotifyAsInactive
+			);
+
+			let round_info = <Round<T>>::get();
+			let max_offline_rounds = T::MaxOfflineRounds::get();
+
+			// Take last round to have rounds_to_check = [7,8,9]
+			// in case we are in round 10 for instance
+			let round = round_info.current.saturating_sub(1);
+
+			let mut rounds_to_check: Vec<RoundIndex> = vec![];
+			for round_index in (0..max_offline_rounds).rev() {
+				rounds_to_check.push(round.saturating_sub(round_index));
```

Comment:
I think this needs a special case for when `round_info.current < max_offline_rounds` because we will end up pushing the same round multiple times to `rounds_to_check` because of `round.saturating_sub(round_index)`. In such a case, we could end up checking the same round multiple times.

I think the extrinsic should fail if `round_info.current < max_offline_rounds` as a pre-check above.

Reply:
Yeah agree on this. Will add the extra check.